package.json

```diff
@@ -63,7 +60,8 @@
     "where": "monorepo-where",
     "reformat": "npm run reformat --workspaces --if-present",
     "prepare": "husky install",
-    "precommit": "precommit"
+    "precommit": "precommit",
+    "preinstall": "node scripts/sort-workspaces.js"
```

Comment:
This will make sure that workspaces are always sorted in topological order according to what's in lerna.json before we even install. Then `npm nun compile --workspaces` (and any npm run --workspaces command) will run in topological order. ie. We don't need lerna in ci for that command.

Why before install? Suppose you add a new package to mongosh. It would still be missing from workspaces if you don't remember to run sort-workspaces first and that will mess with the next install.

That's also why it is a plain javascript script that only uses builtins and not typescript.